superset/dashboards/dao.py

```diff
@@ -54,6 +55,21 @@ def get_datasets_for_dashboard(id_or_slug: str) -> List[Any]:
     def get_charts_for_dashboard(id_or_slug: str) -> List[Slice]:
         return DashboardDAO.get_by_id_or_slug(id_or_slug).slices
 
+    @staticmethod
+    def get_dashboard_changed_on(id_or_slug: str) -> datetime:
```

Comment:
I believe this could also be affected by datasets `changed_on`, but adding that would make this a fairly slow operation to check. The slices `changed_on` check could also be skipped for `/dashboard/:id_or_slug`. Maybe it would be better to have a separate method per endpoint?